src/vl-proza-message.js

```diff
@@ -494,6 +494,21 @@ export class VlProzaMessagePreloader extends vlElement(HTMLElement) {
     });
   }
 
+  /**
+   * Geeft de Proza codes terug op basis van een prefix.
+   *
+   * @param {String} domain - Het Proza domein waarin het Proza bericht zit.
+   * @param {String} prefix - De prefix van de code die het Proza bericht
+   * identificeert.
+   * @return {Promise<[string]>} Resolved naar de Proza codes met de
+   * opgegeven prefix
+   */
+  static getProzaCodes(domain, prefix) {
+    return VlProzaMessagePreloader._getMessages(domain).then((messages) => {
```

Comment:
Dit steunt nu wel op het preloaded zijn van het domein. Als er dus nergens op de pagina een preloader staat, gaat dit niets terug kunnen geven. Je zou evt. ook nog de message cache van VlProzaMessage kunnen gebruiken, daarnaast en/of gewoon alle berichten fetchen uit proza en dat filteren ipv. vanuit een cache te werken.

De methode kan trouwens ook `async` zijn. Dan kan je `await` gebruiken.

Reply:
+1 voor de async methode

Reply:
en idd misschien toch zeker te bekijken of het kan om rekening te houden met het al dan niet al geladen zijn, en als het niet te veel moeite is toch doen

Reply:
Is dat niet juist wat de preloader doet: alle messages halen uit proza? Ik zie niet direct in dat dit zonder zou kunnen werken, zonder een specifieke prefix mee te geven naar de server.

Reply:
@SanderKleykens  moest je dit morgen oppikken bel me anders ff in wil eens de bekijken hoe nu alles binnengetrokken wordt. Ging er o.b.v. de naam nu ook van uit dat de pre-loader de data cachete of iets dergelijks

Reply:
De preloader doet dat, inderdaad, maar dan moet je ook effectief een preloader element gebruiken op de pagina. Als je geen preloader gebruikt hebt voor het domein dat je opvraagt via VlProzaMessagePreloader._getMessages, krijg je een error omdat hij daar niks van preloaded heeft.
De preloader was vooral bedoelt als een manier om messages in bulk op te vragen op een declaratieve manier zodat we niet voor elke <vl-proza-message> een request gingen sturen.

In principe moet je hier dus `VlProzaMessagePreloader._getMessages` oproepen en als er daar niks inzit zou je een fetch moeten doen van alle berichten voor het domein en cachen in de message cache en daar de codes van gebruiken. Je kan dat evt. nog steken in `VlProzaMessagePreloader`, maar eigenlijk is het dan geen preload wat je doet (of is dat toch niet per se de gebruiker van de bedoeling).

Reply:
denk dat gedrag zoals @SanderKleykens het beschrijft voldoende is, dus dat hij ze wel gaat laden, ook al heb je vl-proza-message-preloader element gezet. anders zou je al een call naar de backend moeten voorzien ook, dat lijkt me overkill for now.

Reply:
> @SanderKleykens moest je dit morgen oppikken bel me anders ff in wil eens de bekijken hoe nu alles binnengetrokken wordt. Ging er o.b.v. de naam nu ook van uit dat de pre-loader de data cachete of iets dergelijks

dus ja, die doet dat, maar enkel als je ook een vl-proza-message-preloader ELEMENT ook in je pagina staan hebt ...
denk dat we de case nog moeten dekken waar dat niet het geval zou zijn dus.

Reply:
Heb daarnet even de getProzaCodes methode uitgebreid met een preload en daarvoor ook een nieuwe test case toegevoegd